src/Microsoft.Diagnostics.Monitoring.EventPipe/DiagnosticsEventPipeProcessor.cs

```diff
@@ -69,21 +81,33 @@ public async Task Process(DiagnosticsClient client, int pid, TimeSpan duration,
                     {
                         config = new LoggingSourceConfiguration(_logsLevel);
                     }
-                    if (_mode == PipeMode.Metrics)
+                    else if (_mode == PipeMode.Metrics)
                     {
-                        config = new MetricSourceConfiguration(_metricIntervalSeconds);
+                        config = new MetricSourceConfiguration(_metricIntervalSeconds, _counterFilter.GetProviders());
                     }
-                    if (_mode == PipeMode.GCDump)
+                    else if (_mode == PipeMode.GCDump)
                     {
                         config = new GCDumpSourceConfiguration();
                     }
-                    if (_mode == PipeMode.ProcessInfo)
+                    else if (_mode == PipeMode.ProcessInfo)
                     {
                         config = new SampleProfilerConfiguration();
                     }
+                    else if (_mode == PipeMode.Nettrace)
+                    {
+                        config = _userConfig;
+                    }
 
                     monitor = new DiagnosticsMonitor(config);
                     Stream sessionStream = await monitor.ProcessEvents(client, duration, token);
+
+                    if (_mode == PipeMode.Nettrace)
+                    {
+                        _sessionStarted.TrySetResult(true);
+                        await _streamAvailable(sessionStream, token);
+                        return;
+                    }
+
                     source = new EventPipeEventSource(sessionStream);
```

Comment:
I believe this is not used in the case of PipeMode.Nettrace; could you conditionally construct it, especially since it has the non-zero chance of blocking for a long time?

Reply:
Sorry, I missed that in the case of nettrace that the method returns early.